typescript/sdk/src/deploy/middleware/LiquidityLayerApp.ts

```diff
@@ -110,24 +148,76 @@ export class LiquidityLayerApp<
         txHash,
         message,
         nonce,
-        domain: 0,
+        domain,
         nonceHash: ethers.utils.solidityKeccak256(
           ['uint32', 'uint256'],
-          [0, nonce],
+          [domain, nonce],
         ),
       },
     ];
   }
 
+  async attemptPortalTransferCompletion(
+    message: PortalBridgeMessage<Chain>,
+  ): Promise<void> {
+    const destinationPortalAdapter = this.getContracts(message.destination)
+      .portalAdapter!;
+
+    const transferId = await destinationPortalAdapter.transferId(
+      ChainNameToDomainId[message.origin],
+      message.nonce,
+    );
+
+    const transferMetadata =
+      await destinationPortalAdapter.portalTransfersProcessed(transferId);
+
+    if (transferMetadata.wormholeDomain != 0) {
+      console.log(
+        `Transfer with nonce ${message.nonce} from ${message.origin} to ${message.destination} already processed`,
+      );
+      return;
+    }
+
+    const wormholeOriginDomain = this.config[
+      message.destination
+    ].portal!.wormholeDomainMapping.find(
+      (_) => _.hyperlaneDomain === ChainNameToDomainId[message.origin],
+    )?.wormholeDomain;
+    const emitter = utils.strip0x(
+      utils.addressToBytes32(
+        this.config[message.origin].portal!.portalBridgeAddress,
+      ),
+    );
+
+    const vaa = await fetch(
+      `https://wormhole-v2-testnet-api.certus.one/v1/signed_vaa/${wormholeOriginDomain}/${emitter}/${message.portalSequence}`,
```

Comment:
Consider pulling this base url out to /consts somewhere

Reply:
Also does it need to vary by environment?

Reply:
It does. This was meant as a quick prototype for the hackathons first, we'll need to productionize it for sure in a second iteration